internal/pkg/service/common/etcdop/op/processor.go

```diff
@@ -68,6 +68,30 @@ func (p processors[R]) WithOnResult(fn func(result R)) processors[R] {
 	})
 }
 
+// WithNotEmptyResultAsError is a shortcut for the WithProcessor.
+// If no error occurred yet and the result is NOT an empty value for the R type (nil if it is a pointer),
+// then the callback is executed and returned error is added to the Result.
+func (p processors[R]) WithNotEmptyResultAsError(fn func() error) processors[R] {
+	return p.WithProcessor(func(_ context.Context, result *Result[R]) {
+		emptyValue := result.result == nil || reflect.ValueOf(*result.result).IsZero()
+		emptyValueErr := errors.As(result.Err(), &EmptyResultError{})
+
+		// Empty value is expected, reset EmptyResultError
+		if emptyValueErr {
+			result.ResetErr()
+			return
+		}
```

Comment:
```suggestion
		// Empty value is expected, reset EmptyResultError
		if errors.As(result.Err(), &EmptyResultError{})  {
			result.ResetErr()
			return
		}
```
Cannot it be like that?  I expected that you work with the error afterwards but cannot find any other occurence of `emptyValueErr`

Reply:
:heavy_check_mark: